app/models/data_resources/generic_item.rb

```diff
@@ -0,0 +1,67 @@
+class GenericItem < ApplicationRecord
+  include FilterByRole
+  has_ancestry orphan_strategy: :destroy
+
+  validates_presence_of :generic_type
+  store :payload, coder: JSON
+
+  belongs_to :data_provider
+  has_one :external_reference, as: :external, dependent: :destroy
+  has_many :accessibility_informations, as: :accessable, dependent: :destroy
+  has_many :addresses, as: :addressable, dependent: :destroy
+  has_many :data_resource_categories, as: :data_resource
+  has_many :categories, through: :data_resource_categories
+  has_many :contacts, as: :contactable, dependent: :destroy
+  has_many :companies, as: :companyable, class_name: "OperatingCompany", dependent: :destroy
+  has_many :content_blocks, as: :content_blockable, dependent: :destroy
+  has_many :web_urls, as: :web_urlable, dependent: :destroy
+  has_many :opening_hours, as: :openingable, dependent: :destroy
+  has_many :price_informations, as: :priceable, class_name: "Price", dependent: :destroy
+  has_many :media_contents, as: :mediaable, dependent: :destroy
+  has_many :locations, as: :locateable, dependent: :destroy
```

Comment:
würden wir die hier irgendwie nach einem bestimmten schema sortieren, alphabetisch z.b., wäre das einfacher abzugleichen

Reply:
schwierig: `data_resource_categories` Definition **muss** vor `:categories, through: :data_resource_categories` kommen.

Reply:
dann sollte es aber in jeder datei immer die gleiche reihenfolge haben. sonst vergleicht man sich irre.
oder man hat erst alle `data_resource_...` alphabetisch in einem block und dann alle `...through: :data_resource_...` alphabetisch in einem nächsten block? es gibt doch da best practices für rails?

Reply:
ja, so habe ich es jetzt gelöst gehabt

Reply:
erst die beiden die zusammen gehören und der rest dazu alphabetisch